shell/platform/common/cpp/client_wrapper/texture_registrar_impl.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_CPP_CLIENT_WRAPPER_TEXTURE_REGISTRAR_IMPL_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_CPP_CLIENT_WRAPPER_TEXTURE_REGISTRAR_IMPL_H_
+
+#include "include/flutter/texture_registrar.h"
+
+namespace flutter {
+
+// Wrapper around a FlutterDesktopTextureRegistrarRef that implements the
+// TextureRegistrar API.
+class TextureRegistrarImpl : public TextureRegistrar {
+ public:
+  explicit TextureRegistrarImpl(
+      FlutterDesktopTextureRegistrarRef texture_registrar_ref);
+  virtual ~TextureRegistrarImpl();
+
+  // Prevent copying.
+  TextureRegistrarImpl(TextureRegistrarImpl const&) = delete;
+  TextureRegistrarImpl& operator=(TextureRegistrarImpl const&) = delete;
+
+  int64_t RegisterTexture(TextureVariant* texture) override;
```

Comment:
Flutter engine style is to put a comment on overridden methods that just gives the class that the method is declared it. (In this case there's currently only one, but we try to be consistent about it for convenience of readers and for futureproofing.)